src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs

```diff
@@ -642,18 +636,59 @@ impl HummockLevelsExt for Levels {
             .sum()
     }
 
-    fn apply_compact_ssts(&mut self, summary: GroupDeltasSummary) {
-        let GroupDeltasSummary {
-            delete_sst_levels,
-            delete_sst_ids_set,
-            insert_sst_level_id,
-            insert_sub_level_id,
-            insert_table_infos,
-            ..
-        } = summary;
-
-        if !self.check_deleted_sst_exist(&delete_sst_levels, delete_sst_ids_set.clone()) {
-            warn!(
+    fn apply_commit_epoch(&mut self, group_deltas: &GroupDeltas, has_destroy: bool) {
+        for delta in &group_deltas.group_deltas {
+            if let DeltaType::IntraLevel(intra_level) = delta.get_delta_type().unwrap() {
+                assert!(
+                    intra_level.level_idx == 0 || intra_level.inserted_table_infos.is_empty(),
+                    "we should only add to L0 when we commit an epoch. Inserting into {} {:?}",
+                    intra_level.level_idx,
+                    intra_level.inserted_table_infos
+                );
+                assert!(
+                    intra_level.removed_table_ids.is_empty() || has_destroy,
+                    "no sst should be deleted when committing an epoch"
+                );
+
+                if !intra_level.inserted_table_infos.is_empty() {
+                    insert_new_sub_level(
+                        self.l0.as_mut().unwrap(),
+                        intra_level.l0_sub_level_id,
+                        LevelType::Overlapping,
+                        intra_level.inserted_table_infos.clone(),
+                        None,
+                    );
+                }
+            } else {
+                break;
+            }
+        }
+    }
+
+    fn apply_compact_ssts(&mut self, group_deltas: &GroupDeltas) {
+        let mut has_delete_sst_levels = false;
+        for group_delta in group_deltas.get_group_deltas() {
+            if let DeltaType::IntraLevel(version_delta) = group_delta.get_delta_type().unwrap() {
+                let mut delete_sst_levels = Vec::with_capacity(1);
```

Comment:
Could we collect all insert table and delete sstable ids for every group-delta? 
I means that we can place the variable `delete_sst_levels` outside this loop and change it to `HashMap<u32, HashSet<u64>>`

Reply:
It seems that only group_deltas of the same cg will be submitted each time, so I don't think group_deltas grouping is needed here anymore.

Can we keep it in the current pr and change it after breaking this assumption?